packages/components/message-box/src/index.vue

```diff
@@ -18,7 +18,7 @@
       >
         <el-focus-trap
           loop
-          :trapped="visible"
+          :trapped="autofocus && visible"
```

Comment:
You need to trap irregardless, if autofocus is false set `focus-start-el` to `rootRef` instead of disabling the trap.

Reply:
(Or `focusStartRef` to `rootRef`)

This will resolve the issues of the button colors changing due to focus, while still keeping the dialog accessible. It will not be read out to a screen reader otherwise.